Clarify `prefixUrl` behavior (#943)
sindresorhus/got: readme.md

Type: `string | URL`

> When specified, `prefixUrl` will be prepended to `url`. The prefix can be any valid URL, either relative or absolute. A trailing slash `/` is  - one will be added automatically.

Author: What's missing here?

> When specified, `prefixUrl` will be prepended to `url`. The prefix can be any valid URL, either relative or absolute. A trailing slash `/` is  - one will be added automatically.

Reviewer: > A trailing slash `/` is 

**is** what?

> When specified, `prefixUrl` will be prepended to `url`. The prefix can be any valid URL, either relative or absolute. A trailing slash `/` is  - one will be added automatically.

Author: Oh, thanks, didn't see that one :o

**Note:** `prefixUrl` will be ignored if the `url` argument is a URL instance.

> **Note:** Leading slashes in `input` are disallowed when using this option to enforce consistency and avoid confusion about how the `input` URL is handled, given that `input` will not follow the normal URL resolution rules when `prefixUrl` is being used, which changes the meaning of a leading slash.

Reviewer: I think we could be even more specific and clearer here. Not everyone knows how the URL resolution works. See my text in https://github.com/sindresorhus/got/issues/876#issuecomment-555344208

Author: This was copied directly from Ky :stuck_out_tongue:

Author: ```suggestion
**Note:** Leading slashes in `input` are disallowed when using this option to enforce consistency and avoid confusion. For example, when the prefix URL is `https://example.com/foo` and the input is `/bar`, there's ambiguity whether the resulting URL would become `https://example.com/foo/bar` or `https://example.com/bar`. The latter is used by browsers.
```

> **Tip:** Useful when used with `got.extend()` to create niche-specific Got-instances.

Reviewer: Link to https://github.com/sindresorhus/got#custom-endpoints

> **Tip:** If the input URL still contains the initial `prefixUrl`, you can change it as many times as you want inside hooks. Otherwise it will throw an error.

Reviewer: > If the input URL still contains the initial `prefixUrl`

Not clear to me what this exactly means.

Author: ```suggestion
**Tip:** If the `prefixUrl` is included in the URL, you can change it as many times as you want inside hooks. Otherwise it will throw an error.
```

Author: ```suggestion
**Tip:** You can change `prefixUrl` using hooks as long as the URL includes it. If the URL doesn't include it anymore, it will throw.
```

> **Tip:** You can change `prefixUrl` using hooks as long as the URL includes it. If the URL doesn't include it anymore, it will throw.

Reviewer: ```suggestion
**Tip:** You can change `prefixUrl` using hooks as long as the URL still includes the `prefixUrl`. If the URL doesn't include it anymore, it will throw.
```

?
